src/net.h

```diff
@@ -934,6 +936,10 @@ class CNode
 public:
     uint256 hashContinue;
     std::atomic<int> nStartingHeight{-1};
+    // We selected peer as (compact blocks) high-bandwidth peer (BIP152)
+    bool m_bip152_highbandwidth_to{false};
```

Comment:
As these are mutable fields, add a `GUARDED_BY` annotation?

Reply:
Good point, I'm not sure though on what to put into the annotation, i.e. guarded by _what_? I see three possibilities here to protecting concurrent access to the boolean fields:
- introducing a new Mutex (seems overkill)
- guard with cs_main, as that Mutex is already guaranteed to be held when the flags are written (seems overkill as well, considering this is not validation-critical at all and just about some statistics; locking cs_main in CNode::copyStats() just feels wrong)
- simply using an atomic bool

Any thoughts about this?

Reply:
Atomic bools seem easiest.